GCI-1936: Upgrade mongo driver to use official mongo go driver

VaughanJackson123 opened the pull request:
Update the no longer maintained mgo driver to use the official mongodb driver.

hwilliams1 committed:
Progress to upgrade mongo driver

ethatcher83 committed:
push data to es instance

ethatcher83 committed:
update to push correct size of companies to ES

Commit:
GCI-1936: Clean up code.
- Restore original comments and deferred disconnect call.

dlgroves-ch commented on companybindex/main.go:
```diff
@@ -75,28 +78,41 @@ func main() {
 	f := format.NewFormatter()
-
-	s, err := mgo.Dial(mongoURL)
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(mongoURL))
```
Please use `context.Background()` here.

dlgroves-ch commented on companybindex/main.go:
```diff
@@ -75,28 +78,41 @@ func main() {
+		if err != nil {
+			log.Fatalf("error disconnecting from client: %s", err)
+		}
+	}(client, context.TODO())
```
Please use `context.WithTimeout` with a suitable duration here.

dlgroves-ch commented on companybindex/main.go:
```diff
@@ -75,28 +78,41 @@ func main() {
+	companyProfileCollection := client.Database(mongoDatabase).Collection(mongoCollection)
+	findOptions := options.Find()
+	findOptions.SetBatchSize(int32(mongoSize))
+	cur, err := companyProfileCollection.Find(context.TODO(), bson.D{}, findOptions)
```
Please use `context.WithTimeout` with a suitable duration here.

dlgroves-ch commented on companybindex/main.go:
```diff
@@ -75,28 +78,41 @@ func main() {
-			result := datastructures.MongoCompany{}
-
-			if !it.Next(&result) {
+			if !cur.Next(context.TODO()) {
```
Please use `context.WithTimeout` with a suitable duration here.

Commit:
GCI-1936: Address PR Review comments.
- Use context.Background() for initial context.
- Use context.WithTimeout() for operations that might hang.
